### ISSUE TITLE
Use with Activeform

### ISSUE THREAD

**Issue by dbd5** [alfa6661/yii2-raty#1]
Hello,

How do you use this with activeform and a model?

Regards,

Adam

**Comment by dbd5** [comment]
Hi Alfa666,

Seems there is an issue, I get the following error after I setup the widget

PHP Fatal Error – yii\base\ErrorException
Class name must be a valid object or a string
$this->parts['{input}'] = $class::widget($config);

**Comment by alfa6661** [comment]
what is $class?
can you post the complete code?

**Comment by dbd5** [comment]
Here's the code below from my _form. 
It's currently using the StarRating widget by Kartik. 
The error occurs once I replace the StarRating config with Raty config you provided

<?php

use yii\helpers\Html;
use yii\widgets\ActiveForm;
use yii\helpers\ArrayHelper;
use common\models\Product;
use kartik\rating\StarRating;

/\* @var $this yii\web\View _/
/_ @var $model common\models\Feedback _/
/_ @var $form yii\widgets\ActiveForm _/
/_ @var $c _/
/_ @var $p _/
/_ @var $t */

?>

<div class="feedback-form">
    <?php $form = ActiveForm::begin(); ?>
    <?php // echo $form->field($model, 'user_id')->textInput() ?>
    <div class="row">
        <?php // echo $form->field($model, 'product_id')->textInput() ?>
        <div class="col-lg-6">
            <?php
                if ($model->isNewRecord) {
                    $model->product_id = $p;
                    echo $form->field($model, 'product_id')->dropDownList(
                        ArrayHelper::map(product::find()->all(), 'id', 'name'),
                        ['Prompt' => 'Select product']
                    );
                } else {
                    echo $form->field($model, 'product_id')->dropDownList(
                        ArrayHelper::map(product::find()->all(), 'id', 'name'),
                        ['Prompt' => 'Select product', ['disabled' => true]]
                    );
                }
            ?>
        </div>
        <?php // echo $form->field($model, 'type')->textInput() ?>
        <div class="col-lg-3">
            <?php
                if ($model->isNewRecord) {
                    $model->type = $t;
                }
                echo $form->field($model, 'type')->dropDownList($model->typeList);
            ?>
        </div>
        <?php // echo $form->field($model, 'time')->textInput() ?>
        <div class="col-lg-3">
            <?= $form->field($model, 'status')->dropDownList($model->statusList) ?>
        </div>
    </div>
    <div class="row">
        <?php // echo $form->field($model, 'detail')->textInput(['maxlength' => true]) ?>
        <div class="col-lg-6">
            <?= $form->field($model, 'detail')->textarea(['rows' => 12]) ?>
        </div>
        <div class="col-lg-6">
            <?= $form->field($model, 'amount')->textInput(['maxlength' => true]) ?>
        </div>
        <?php // echo $form->field($model, 'required')->textInput(['maxlength' => true]) ?>
        <div class="col-lg-6">
            <?= $form->field($model, 'required')->textarea(['rows' => 4]) ?>
        </div>
        <div class="col-lg-6">
            <?php
            echo $form->field($model, 'rating')->widget(StarRating::classname(), [
                'pluginOptions' => [
                    'glyphicon' => false,
                    'size' => 'sm', // xl, lg, md, sm, and xs.
                    'showCaption' => true,
                    'showClear' => true,
                ],
            ]);
            ?>
       </div>
    </div>
    <?php // echo $form->field($model, 'status')->textInput() ?>
    <?php // echo $form->field($model, 'visibility')->textInput() ?>
    <?php // echo $form->field($model, 'created_at')->textInput() ?>
    <?php // echo $form->field($model, 'updated_at')->textInput() ?>
    <div class="form-group">
        <?= Html::submitButton($model->isNewRecord ? Yii::t('app', 'Create') : Yii::t('app', 'Update'), ['class' => $model->isNewRecord ? 'btn btn-success' : 'btn btn-primary']) ?>
    </div>
    <?php ActiveForm::end(); ?>
</div>

**Comment by alfa6661** [comment]
StarRating widget and Raty are different. StarRating widget by kartik has different configurations with Raty. You can't replace the StarRating config with Raty config (see https://github.com/wbotelhos/raty#options).

if you want to you use Raty.
try using below code:

``` php
<?php
use alfa6661\widgets\Raty;
echo $form->field($model, 'rating')->widget(Raty::className()) ?>
```

**Comment by dbd5** [comment]
Hi Alfa666,

The intent is not to replace the StarRating config with Raty config.
The example you provided above will work, I'm just pointing out that the one with config provided in your docs as reproduced below will NOT work due to a syntax error. Drop it in any form and see.

Suggest you correct the docs with correct syntax for the Activeform and Model use case

Regards

// Usage with ActiveForm and model
<?= $form->field($model, 'rating')->widget(\alfa6661\widgets\Raty::className()
    'options' => [
        // the HTML attributes for the widget container
    ],
    'pluginOptions' => [
        // the options for the underlying jQuery Raty plugin
        // see : https://github.com/wbotelhos/raty#options
    ]
) ?>

**Comment by alfa6661** [comment]
ah sorry, i was wrong.
i have fix the docs: 59e1c85943ba8e3e1b4e183e8ccd6aad38823151

thanks.

**Comment by dbd5** [comment]
Great work Bro!

U're welcome!

Adam
